v0.6.0

jlmaurer opened the pull request:
## [0.6.0]

### Removed
* [764](https://github.com/dbekaert/RAiDER/pull/764) - Removed Python 3.8 support. Python 3.9 is now the minimum version officially required to run RAiDER.
* [753](https://github.com/dbekaert/RAiDER/pull/753) - Removed support for ERA-Interim.

### Changed
* [743](https://github.com/dbekaert/RAiDER/pull/743) - Switched from HTTPS to DAP4 for retrieving MERRA2 data, and suppressed a warning for using DAP4 for GMAO data where doing so is not possible.

### Fixed
* [765](https://github.com/dbekaert/RAiDER/pull/765) - Fixed a dead link to "Installing from Source" in the ReadMe.
* [759](https://github.com/dbekaert/RAiDER/pull/759) - Added a `browse` S3 tag for `.png` files when uploaded to AWS.
* [751](https://github.com/dbekaert/RAiDER/pull/751) - Fixed ERA-5 interface for a change to its API that requires pressure variables to be requested separately from temperature and humidity.
* [741](https://github.com/dbekaert/RAiDER/pull/741) - Updated mamba setup commands in CircleCI for mamba 2.0.0.
* [738](https://github.com/dbekaert/RAiDER/pull/738) - Resolved an InsecureRequestWarning when fetching data through CDS, such as for ERA5.
* [734](https://github.com/dbekaert/RAiDER/pull/734) - Resolved a deprecation warning from Rasterio about the use of `statistics()`.
* [733](https://github.com/dbekaert/RAiDER/pull/733) - Fixed coverage reports being generated for incorrect directories.
* [732](https://github.com/dbekaert/RAiDER/pull/732) - Updated `hrrr.py` to resolve a deprecation warning about `unary_union`.
* [731](https://github.com/dbekaert/RAiDER/pull/731) - Fixed fetch routine for GMAO.

### Added
* [761](https://github.com/dbekaert/RAiDER/pull/761) - Added Python 3.13 support.
* [762](https://github.com/dbekaert/RAiDER/pull/762) - Added lockfiles through conda-lock to the repo, documentation, and CircleCI runners.
* [747](https://github.com/dbekaert/RAiDER/pull/747) - Added support for run config time format `%H:%M:%S` (without quotes).
* [746](https://github.com/dbekaert/RAiDER/pull/746) - Added support for numpy v2.
* [725](https://github.com/dbekaert/RAiDER/pull/725) - Added rules to ignore all test artifacts in git.
* [728](https://github.com/dbekaert/RAiDER/pull/728) - Added downloader tests for HRES, GMAO, and MERRA2.
* [726](https://github.com/dbekaert/RAiDER/pull/726) - Updated code and documentation for changes to the CDS API.

dependabot[bot] committed:
Bump ASFHyP3/actions from 0.17.1 to 0.20.0

Bumps [ASFHyP3/actions](https://github.com/asfhyp3/actions) from 0.17.1 to 0.20.0.
- [Release notes](https://github.com/asfhyp3/actions/releases)
- [Changelog](https://github.com/ASFHyP3/actions/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/asfhyp3/actions/compare/v0.17.1...v0.20.0)

---
updated-dependencies:
- dependency-name: ASFHyP3/actions
  dependency-version: 0.20.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

jhkennedy committed:
update actions with permissions, switch to setup-micromamba action,
updated usages for actions

jhkennedy committed:
label dependabot prs bumpless

jhkennedy committed:
Merge pull request #724 from dbekaert/dependabot/github_actions/ASFHyP3/actions-0.20.0

Bump ASFHyP3/actions from 0.17.1 to 0.20.0

nate-kean committed:
Ignore more temp test artifacts

This also deletes an artifact that I accidentally committed a year ago >_<

nate-kean committed:
Add trailing newline

nate-kean committed:
Update CHANGELOG.md for #725

mohseniaref committed:
Fix look direction assertion

jlmaurer committed:
Merge pull request #725 from garlic-os/ignore-test-artifacts

Ignore temporary test artifacts

mohseniaref committed:
Fix nested list creation when parsing station file

mohseniaref committed:
Fix typo in WeatherModels documentation

mohseniaref committed:
Merge pull request #5 from mohseniaref/codex/fix-typo-in-weathermodels.md

Fix typo in WeatherModels documentation

mohseniaref committed:
Fix nested list creation when parsing station file

mohseniaref committed:
Merge pull request #6 from mohseniaref/enpbd7-codex/fix-data-appending-in-downloadgnssdelays.py

Fix reading of whitespace-separated station files

mohseniaref committed:
Fix nested list creation when parsing station file

mohseniaref committed:
Fix typo 'occurence' to 'occurrence'

mohseniaref committed:
Merge pull request #8 from mohseniaref/codex/fix-spelling-of-occurrence-in-code

Fix typos in comments

mohseniaref committed:
Merge pull request #3 from mohseniaref/codex/find-and-fix-a-bug-in-codebase

Fix look direction assertion bug

mohseniaref committed:
Merge pull request #7 from mohseniaref/18gix2-codex/fix-data-appending-in-downloadgnssdelays.py

Fix reading of whitespace-separated station files

mohseniaref committed:
Merge pull request #4 from mohseniaref/codex/fix-data-appending-in-downloadgnssdelays.py

Fix reading of whitespace-separated station files

nate-kean committed:
Incremental linting, formatting

Some linting and formatting changes throughout the project as I go.
- test\test_downloaders.py: added type annotations, swapped os.path for pathlib
- test\test_weather_model.py: added type annotations
- tools\RAiDER\models\ecmwf.py:
  - replaced calls to module-level strftime() with more succinct calls to datetimes/times' strftime() method
  - Added type annotations to _get_from_cds()
- tools\RAiDER\models\weatherModel.py: added type annotation for WeatherModel._model_level_type
- tools\RAiDER\utilFcns.py: fixed incorrect type annotation on round_date()'s "precision" argument

nate-kean committed:
Merge branch 'dev' of https://github.com/dbekaert/RAiDER into dev

nate-kean committed:
Update access instructions for ECMWF API changes

- Fixed dead links to registration and ERA5 API configuration pages.
- Changed wording to refer to "products" as "datasets", to align with the terminology used on ECMWF's websites
- Several grammatical/formatting corrections

nate-kean committed:
Add note to existing users about CDS API changes

nate-kean committed:
Update CHANGELOG.md for #726

nate-kean committed:
Update CDS API help URL

nate-kean committed:
Update CDS API URL

nate-kean committed:
typo

nate-kean committed:
Add warning for old CDS API URL, + test

nate-kean committed:
Use TypeVar for older Python versions

nate-kean committed:
Update CDS API URL

nate-kean committed:
Add downloader tests for HRES, GMAO, MERRA2

nate-kean committed:
Update CHANGELOG.md for #728

jlmaurer committed:
Merge pull request #726 from garlic-os/cds-api-update

Update RAiDER for CDS API changes

jlmaurer committed:
Merge branch 'dev' into more-downloader-tests

jlmaurer committed:
Merge pull request #728 from garlic-os/more-downloader-tests

Add downloader tests for HRES, GMAO, MERRA2

nate-kean committed:
Fix #664: replace geopandas unary_union with union_all()

nate-kean committed:
Fix types for writeWeatherVarsXarray

nate-kean committed:
Fix incomplete _fetch function in GMAO

nate-kean committed:
GMAO._fetch(): don't suppress exceptions

nate-kean committed:
Use new dir for each test run, format

- Formatting
- Repeated test code parametrized
- Temporary directory use for each run of a test that subsequent test runs don't short circuit ("there's no work to do") on a file that already exists

nate-kean committed:
Get pydap credentials from .netrc

Affects the two NASA EarthData models: GMAO and MERRA2

This also obsolesces read_EarthData_loginInfo

nate-kean committed:
WeatherModels' _fetch(): use Path

nate-kean committed:
Update CHANGELOG.md for #731

nate-kean committed:
Update CHANGELOG.md for #732

nate-kean committed:
Import importlib.util to use importlib.util.find_spec

nate-kean committed:
Update CHANGELOG.md for dbekaert#733

nate-kean committed:
Update setup_from_env() for CDS API update

- Update .cdsapirc templates for the CDS API update
- Don't check for UID if the API doesn't use it

nate-kean committed:
Ignore generated coverage files

nate-kean committed:
Fix "CoverageWarning: No data was collected"

I regret to say, but I don't know why this fixes it. I just know that this is another way to do it when pytest-cov is installed, and in RAiDER it is. I threw stuff at the wall and this stuck.

mohseniaref committed:
Merge branch 'dev' into dev

nate-kean committed:
Fix Rasterio statistics() deprecation warning

nate-kean committed:
Update CHANGELOG.md for #734

nate-kean committed:
Fix CDS InsecureRequestWarning

nate-kean committed:
Update CHANGELOG.md for #738

nate-kean committed:
CircleCI "Create RAiDER env": increase timeout

nate-kean committed:
Update CHANGELOG.md for #741

nate-kean committed:
Enable xtrace on Install MiniForge

nate-kean committed:
(cleanup) slight spelling corrections

nate-kean committed:
Use alternative to mamba.sh

nate-kean committed:
Merge branch 'fix-coveralls' into fix-mamba-sh-warning-2

nate-kean committed:
Merge branch 'fix-gmao-fetch' into suppress-dap-https-warning

nate-kean committed:
GMAO: Suppress pydap warning

nate-kean committed:
MERRA2: Use DAP4 protocol

jlmaurer committed:
keep earthdata creds testing

jlmaurer committed:
Restore read_EarthData_loginInfo

jlmaurer committed:
Keep earthdata check fcn

nate-kean committed:
Revert "Restore read_EarthData_loginInfo"

This reverts commit ceb18fe44b83721971a4af351d62750b30bc044e.

nate-kean committed:
Merge branch 'fix-gmao-fetch' of https://github.com/garlic-os/RAiDER into fix-gmao-fetch

nate-kean committed:
Revert "Merge branch 'fix-gmao-fetch' of https://github.com/garlic-os/RAiDER into fix-gmao-fetch"

This reverts commit 3a0f4de9b88e94af74d788c9cfac595a7ed6a50d, reversing
changes made to 21b9eab62f30e73ab38c2a20475d89a1b8167d48.

jlmaurer committed:
Merge pull request #731 from garlic-os/fix-gmao-fetch

Fix GMAO downloader

jlmaurer committed:
Merge branch 'dev' into fix-664

jlmaurer committed:
Merge pull request #732 from garlic-os/fix-664

Fix #664: replace geopandas unary_union with union_all()

jlmaurer committed:
Merge branch 'dev' into fix-deprecation-rasterio-statistics

jlmaurer committed:
Merge pull request #734 from garlic-os/fix-deprecation-rasterio-statistics

Fix Rasterio statistics() deprecation warning

jlmaurer committed:
Merge pull request #727 from mohseniaref/dev

Fixing minor type and a nested list

jlmaurer committed:
Merge branch 'dev' into fix-coveralls

jlmaurer committed:
Merge pull request #733 from garlic-os/fix-coveralls

Fix Coveralls

nate-kean committed:
Merge branch 'dev' into fix-cds-InsecureRequestWarning

nate-kean committed:
Set minimum numpy version to 2.0

nate-kean committed:
Add ruff ruleset for numpy 2.0 migration

nate-kean committed:
Migrate np.trapz to np.trapezoid

jlmaurer committed:
Merge pull request #738 from garlic-os/fix-cds-InsecureRequestWarning

Fix CDS InsecureRequestWarning

jlmaurer committed:
Merge branch 'dev' into fix-mamba-sh-warning-2

nate-kean committed:
Update CHANGELOG.md for #746

jlmaurer committed:
Merge pull request #742 from garlic-os/fix-mamba-sh-warning-2

Fix #740: Use alternative to mamba.sh to maintain compatibility with mamba 2.0.0

nate-kean committed:
Merge branch 'dbekaert:dev' into fix-dap-https-warning

nate-kean committed:
Update CHANGELOG.md for #743

jlmaurer committed:
Merge pull request #743 from garlic-os/fix-dap-https-warning

Use DAP4 where possible, suppress warning otherwise

nate-kean committed:
Interpret int time input as timestamp

nate-kean committed:
Update CHANGELOG.md for #747

nate-kean committed:
Add test for int time input

nate-kean committed:
Support both numpy v1 and v2

- environment.yml: Remove requirement for numpy to be at least v2
- Add polyfill for np.trapezoid for numpy v1

jlmaurer committed:
Merge pull request #746 from garlic-os/numpy-2.0-upgrade

Upgrade to numpy 2.0

nate-kean committed:
Remove ERA-Interim

nate-kean committed:
Remove ERA-Interim

nate-kean committed:
Update CHANGELOG.md for #753

nate-kean committed:
Update CHANGELOG.md for #753

jacquelynsmale committed:
add in get_tag_set to aws.py

jacquelynsmale committed:
fix changelog to reflect tag

jacquelynsmale committed:
switch to accept only Path

jacquelynsmale committed:
fix path in test

jacquelynsmale committed:
fix patch side effect

jacquelynsmale committed:
fix 135

cmarshak committed:
Merge pull request #759 from jacquelynsmale/add_browse_s3_tag

Add `browse` s3 tag to  uploaded `.png` files

cmarshak committed:
Merge pull request #760 from dbekaert/herman-smale-branch

Browse Imagery Fix

nate-kean committed:
Fix tests: use xarray for MERRA2._fetch()

nate-kean committed:
Fix test_temporal_interpolate.py: use xarray for MERRA2._fetch()

nate-kean committed:
Update CHANGELOG.md for #761

nate-kean committed:
Generate lockfile for Python 3.12 @ 5727651

nate-kean committed:
Generate lockfile for Python 3.12 @ 5727651

nate-kean committed:
Update documentation for lockfiles

nate-kean committed:
Update documentation for lockfiles

nate-kean committed:
Update CHANGELOG.md for #762

nate-kean committed:
Update CHANGELOG.md for #762

nate-kean committed:
Create regenerate-locks.py

nate-kean committed:
Create regenerate-locks.py

nate-kean committed:
Update CircleCI config to use lockfiles

nate-kean committed:
Update CircleCI config to use lockfiles

nate-kean committed:
Generate lockfiles for all other supported Python versions @ 5727651

nate-kean committed:
Generate lockfiles for all other supported Python versions @ 5727651

jlmaurer committed:
Merge pull request #762 from nate-kean/add-lock-file

Add lockfiles to RAiDER

dependabot[bot] committed:
Bump actions/checkout from 4 to 5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 4 to 5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

nate-kean committed:
Merge branch 'dev' into remove-erai

Squashed commit of the following:

commit b32655c0b28376e03fad96795b697a8c120fdd63
Merge: 5727651 3fd5cf5
Author: Jeremy Maurer <maurer.jeremy@gmail.com>
Date:   Sat Aug 23 21:24:38 2025 -0500

    Merge pull request #762 from nate-kean/add-lock-file

    Add lockfiles to RAiDER

commit 3fd5cf5b786351dc86d7e6d494b42ee17d2d7f98
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:20:28 2025 -0500

    Create regenerate-locks.py

commit a999d56edaeaea3dc799d14b2d56ec2a04826167
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:15:06 2025 -0500

    Update CHANGELOG.md for #762

commit 56df113e35a3cc1d5efb9fe671b7c8785a0d5871
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:10:52 2025 -0500

    Update documentation for lockfiles

commit d80989253e4bb6c8d8ab4356a577ddfa4afe73a1
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:32:32 2025 -0500

    Update CircleCI config to use lockfiles

commit d58b08be5de0b21562577473b438b14c02f085ca
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:32:59 2025 -0500

    Generate lockfiles for all other supported Python versions @ 5727651

commit a78ec5af38ee903ccc98b8afac5493066d4f0968
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:09:05 2025 -0500

    Generate lockfile for Python 3.12 @ 5727651

commit 0bb20687da7df17c01e998ba13c548a182f177ae
Author: Nate Kean <natekean65535@gmail.com>
Date:   Thu Aug 21 20:36:44 2025 -0500

    Fix tests: use xarray for MERRA2._fetch()

commit 57276516bacf6596d252eb4f63f16aeaceb68120
Merge: 784237b 6d12fd4
Author: Charlie Marshak <cmarshak@users.noreply.github.com>
Date:   Thu Aug 21 13:11:09 2025 -0700

    Merge pull request #760 from dbekaert/herman-smale-branch

    Browse Imagery Fix

commit 6d12fd40f85960a89e99f611988d74a768ee0413
Merge: 784237b a1d5d3c
Author: Charlie Marshak <cmarshak@users.noreply.github.com>
Date:   Thu Aug 21 11:38:18 2025 -0700

    Merge pull request #759 from jacquelynsmale/add_browse_s3_tag

    Add `browse` s3 tag to  uploaded `.png` files

commit a1d5d3c87bb407a234cc9731a644fe9c71489d1f
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 16:29:46 2025 -0800

    fix 135

commit 0ee91abd92cbaa44db51f75040da04164281a7fe
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 16:09:46 2025 -0800

    fix patch side effect

commit c7db467eb83ed7a097b034bdb28fc277e32071f3
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 15:12:35 2025 -0800

    fix path in test

commit 4704db3af6cd3070cd2fe959fa36830ed9ed317b
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 13:52:37 2025 -0800

    switch to accept only Path

commit 5f929ba5cff5ddf4049f5f0a06681b84ea0170eb
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Tue Aug 19 15:46:07 2025 -0800

    fix changelog to reflect tag

commit 1a16ac8d2c075ade59b0db4d125b63c421cfa503
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Tue Aug 19 15:36:23 2025 -0800

    add in get_tag_set to aws.py

nate-kean committed:
Merge branch 'dev' into accept-yaml-int-timestamp-format

Squashed commit of the following:

commit b32655c0b28376e03fad96795b697a8c120fdd63
Merge: 5727651 3fd5cf5
Author: Jeremy Maurer <maurer.jeremy@gmail.com>
Date:   Sat Aug 23 21:24:38 2025 -0500

    Merge pull request #762 from nate-kean/add-lock-file

    Add lockfiles to RAiDER

commit 3fd5cf5b786351dc86d7e6d494b42ee17d2d7f98
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:20:28 2025 -0500

    Create regenerate-locks.py

commit a999d56edaeaea3dc799d14b2d56ec2a04826167
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:15:06 2025 -0500

    Update CHANGELOG.md for #762

commit 56df113e35a3cc1d5efb9fe671b7c8785a0d5871
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:10:52 2025 -0500

    Update documentation for lockfiles

commit d80989253e4bb6c8d8ab4356a577ddfa4afe73a1
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:32:32 2025 -0500

    Update CircleCI config to use lockfiles

commit d58b08be5de0b21562577473b438b14c02f085ca
Author: Nate Kean <natekean65535@gmail.com>
Date:   Sat Aug 23 16:32:59 2025 -0500

    Generate lockfiles for all other supported Python versions @ 5727651

commit a78ec5af38ee903ccc98b8afac5493066d4f0968
Author: Nate Kean <natekean65535@gmail.com>
Date:   Fri Aug 22 16:09:05 2025 -0500

    Generate lockfile for Python 3.12 @ 5727651

commit 0bb20687da7df17c01e998ba13c548a182f177ae
Author: Nate Kean <natekean65535@gmail.com>
Date:   Thu Aug 21 20:36:44 2025 -0500

    Fix tests: use xarray for MERRA2._fetch()

commit 57276516bacf6596d252eb4f63f16aeaceb68120
Merge: 784237b 6d12fd4
Author: Charlie Marshak <cmarshak@users.noreply.github.com>
Date:   Thu Aug 21 13:11:09 2025 -0700

    Merge pull request #760 from dbekaert/herman-smale-branch

    Browse Imagery Fix

commit 6d12fd40f85960a89e99f611988d74a768ee0413
Merge: 784237b a1d5d3c
Author: Charlie Marshak <cmarshak@users.noreply.github.com>
Date:   Thu Aug 21 11:38:18 2025 -0700

    Merge pull request #759 from jacquelynsmale/add_browse_s3_tag

    Add `browse` s3 tag to  uploaded `.png` files

commit a1d5d3c87bb407a234cc9731a644fe9c71489d1f
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 16:29:46 2025 -0800

    fix 135

commit 0ee91abd92cbaa44db51f75040da04164281a7fe
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 16:09:46 2025 -0800

    fix patch side effect

commit c7db467eb83ed7a097b034bdb28fc277e32071f3
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 15:12:35 2025 -0800

    fix path in test

commit 4704db3af6cd3070cd2fe959fa36830ed9ed317b
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Wed Aug 20 13:52:37 2025 -0800

    switch to accept only Path

commit 5f929ba5cff5ddf4049f5f0a06681b84ea0170eb
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Tue Aug 19 15:46:07 2025 -0800

    fix changelog to reflect tag

commit 1a16ac8d2c075ade59b0db4d125b63c421cfa503
Author: jacquelynsmale <jacquelyn.smale@gmail.com>
Date:   Tue Aug 19 15:36:23 2025 -0800

    add in get_tag_set to aws.py

commit 784237b7e0a94c7748ae0e421636d7e4830a1a28
Merge: 9eecf23 a259c66
Author: Jeremy Maurer <maurer.jeremy@gmail.com>
Date:   Thu Jul 3 20:38:01 2025 -0500

    Merge pull request #746 from garlic-os/numpy-2.0-upgrade

    Upgrade to numpy 2.0

commit a259c669d6092520db91917608ba7c64f17283c1
Author: Nate Kean <14845347+garlic-os@users.noreply.github.com>
Date:   Wed Jul 2 17:56:17 2025 -0500

    Support both numpy v1 and v2

    - environment.yml: Remove requirement for numpy to be at least v2
    - Add polyfill for np.trapezoid for numpy v1

commit 4a5d4ea0bb443c580fdd8bc986c7d7485385f3ff
Author: Nate Kean <14845347+garlic-os@users.noreply.github.com>
Date:   Mon Jun 30 21:14:40 2025 -0500

    Update CHANGELOG.md for #746

commit 46e88836a82a543685193051e3da53bcd8fdb9cd
Author: Nate Kean <14845347+garlic-os@users.noreply.github.com>
Date:   Mon Jun 30 21:07:39 2025 -0500

    Migrate np.trapz to np.trapezoid

commit 7d3c98c043c4f735e9a00b58f195dbcf9b730be3
Author: Nate Kean <14845347+garlic-os@users.noreply.github.com>
Date:   Mon Jun 30 21:06:01 2025 -0500

    Add ruff ruleset for numpy 2.0 migration

commit b5e518dd7717280a50b3cefd6ddb45369f116859
Author: Nate Kean <14845347+garlic-os@users.noreply.github.com>
Date:   Mon Jun 30 21:05:45 2025 -0500

    Set minimum numpy version to 2.0

nate-kean committed:
Merge branch 'dbekaert:dev' into remove-erai

jlmaurer committed:
Merge pull request #758 from dbekaert/dependabot/github_actions/actions/checkout-5

Bump actions/checkout from 4 to 5

nate-kean committed:
Update regenerate-locks.py

- Reduce data duplication: Pull Python versions from .circleci/config.yaml instead of hardcoding them again
- Improve logging
- Add comments, documentation

jlmaurer committed:
Merge pull request #753 from nate-kean/remove-erai

Remove ERA-Interim

nate-kean committed:
Merge branch 'dev' into accept-yaml-int-timestamp-format

nate-kean committed:
Add Python 3.13 support, set as highest supported version

nate-kean committed:
Update CHANGELOG.md for #761

nate-kean committed:
Remove Python 3.8 support

- Update mentions of 3.8 (or 3.7) as minimum version to say 3.9

nate-kean committed:
Update CHANGELOG.md for #764

nate-kean committed:
Merge branch 'dev' into remove-python-3.8

nate-kean committed:
Fix dead link "Installing from Source"

nate-kean committed:
Update CHANGELOG.md for #765

nate-kean committed:
Format

nate-kean committed:
Fix test_GUNW_hyp3_metadata_update: fix invalid gunw_schema.json

nate-kean committed:
Merge branch 'dev' into test-fixes

jlmaurer committed:
Merge pull request #747 from nate-kean/accept-yaml-int-timestamp-format

Interpret int time input as timestamp

jlmaurer committed:
Merge pull request #764 from nate-kean/remove-python-3.8

Remove Python 3.8 support

jlmaurer committed:
Merge pull request #765 from nate-kean/patch-1

Fix dead link "Installing from Source"

nate-kean committed:
Merge branch 'dev' into test-fixes

nate-kean committed:
Upgrade jsonchema from ==3.2.0 to >=4

nate-kean committed:
Use fuzzy = instead of strict ==

jlmaurer committed:
Merge pull request #761 from nate-kean/test-fixes

Add Python 3.13 compatibility

nate-kean committed:
Update CHANGELOG.md for #751

nate-kean committed:
Rename warpedDEM.dem to .rdr

These are actually height files in RADAR coordinates, not DEMs in satellite coordinates. "A DEM that has been warped into a height file". This makes that more clear by giving them the file extension typically used for files in RADAR coordinates.

nate-kean committed:
Cherry-pick test_intersect.py from c1b6c21

nate-kean committed:
Update test_scenario_1

- Retooled to work with the current RAiDER API
- The test in test_HRRR_ztd.py had to be renamed to prevent a name collision

nate-kean committed:
Fix: Finish plumbing out dir arg to AOI classes

nate-kean committed:
Add/fix type annotations

Introduces a way to annotate the shape of numpy arrays, which can be extremely helpful for understanding and debugging the areas that do lots of work on arrays.

Add/fix type annotations

Add/fix type annotations

merge this into a delay.py commit

nate-kean committed:
Update for ERA-5 API changes

- Reworked height calculation for ECMWF's new method
- Reworked ECMWF._fetch() for their API changes: now makes two requests and calculates out the missing heights before saving
- Formatting on ECMWF._load_model_levels()

nate-kean committed:
Add ERA-5 to scenario_4 test

nate-kean committed:
Use raise-from for better debugging

jlmaurer committed:
Merge pull request #751 from nate-kean/fix-for-era5-api-change

Fix for ERA-5 API change

jlmaurer committed:
Update CHANGELOG.md

jlmaurer commented:
I'm calling this a minor release because we dropped support for Python 3.8 and added 3.13, and we've removed ERA-I support.

jlmaurer commented:
I think this is ready once #767 is merged.

cmarshak committed:
Merge pull request #767 from dbekaert/jlmaurer-patch-2

Update CHANGELOG.md

cmarshak reviewed (approved):
Looks fine with the changelog updated.

nate-kean committed:
Fix type annotations

nate-kean committed:
Format

- Sort imports
- import datetime as dt (reduce ambiguity between datetime and datetime.datetime)
- Linting: add closing punctuation to first line of docstrings
- Remove superfluous `assert True` statements

nate-kean committed:
ruff format

Plus manual overrides for blocks that break usual formatting rules intentionally

jlmaurer committed:
Merge pull request #768 from nate-kean/fix-test_hrrrak_dl

Lint/format test_weather_model.py
